### PR TITLE
Stop supporting RuntimeConfig in v2

### PR DESCRIPTION
**Pull request by inlined**
Stop uploading .runtimeconfig.json for new functions APIs.

### CODE REVIEW

File: src/deploy/functions/prepareFunctionsUpload.ts [hunk 1]
```diff
@@ -90,10 +90,12 @@ async function packageSource(options: Options, sourceDir: string, configValues:
         mode: file.mode,
       });
     });
-    archive.append(JSON.stringify(configValues, null, 2), {
-      name: CONFIG_DEST_FILE,
-      mode: 420 /* 0o644 */,
-    });
+    if (typeof configValues !== "undefined") {
```
Comment (taeold) [inline src/deploy/functions/prepareFunctionsUpload.ts:93]: q: Is this equivalent to:

```js
if (configValues !== undefined) {
...
```

Is there reason to prefer one over another?

File: src/deploy/functions/prepare.ts [hunk 2]
```diff
@@ -80,7 +80,15 @@ export async function prepare(
       clc.bold(options.config.src.functions.source) +
       " directory for uploading..."
   );
-  context.functionsSource = await prepareFunctionsUpload(runtimeConfig, options);
+  if (wantBackend.cloudFunctions.find((fn) => fn.platform === "gcfv1")) {
+    context.functionsSourceV1 = await prepareFunctionsUpload(runtimeConfig, options);
+  }
+  if (wantBackend.cloudFunctions.find((fn) => fn.platform === "gcfv2")) {
+    context.functionsSourceV2 = await prepareFunctionsUpload(
```
Comment (taeold) [inline src/deploy/functions/prepare.ts:87]: Is there an opportunity for us to inform the user that runtimeconfig will be ignored for GCFv2 deploy?

**Review by taeold — APPROVED** [review]
LGTM, 1 comment and 1 question.

File: src/deploy/functions/prepareFunctionsUpload.ts (hunk 1 above)
Comment (inlined) [reply src/deploy/functions/prepareFunctionsUpload.ts:93]: Though it doesn't seem to work in new versions of Node, I was raised in a world where `undefined` was a variable and not a literal, so you were always supposed to check the type of a value and not compare it to the variable `undefined`.

File: src/deploy/functions/prepare.ts (hunk 2 above)
Comment (inlined) [reply src/deploy/functions/prepare.ts:87]: I don't think so? It's perfectly valid, though wonky, to use config in your v1 functions and not in your v2 functions. And I don't recall your migrate script deleting config either. Do you have a good way to decide when to warn?

**Pull request merged**